### PR TITLE
check authorization on createMessage

### PR DESCRIPTION
**Pull request by Schparky**

### CODE REVIEW

File: application/models/post.go [hunk 1]
```diff
@@ -759,8 +759,16 @@ func (p *Post) FindByUserAndUUID(ctx context.Context, user User, uuid string) er
 		Where("uuid = ?", uuid).First(p)
 }
 
-// FindByUser finds all posts visible to the current user.
-func (p *Posts) FindByUser(ctx context.Context, user User, destination, origin *Location, searchText *string) error {
+// PostFilterParams are optional parameters to narrow the list of posts returned from a query
+type PostFilterParams struct {
```
Comment (Baggerone) [inline application/models/post.go]: Good addition.

File: application/models/post.go [hunk 2]
```diff
@@ -793,23 +801,27 @@ func (p *Posts) FindByUser(ctx context.Context, user User, destination, origin *
 	args := []interface{}{user.ID, PostVisibilityAll, PostVisibilityTrusted, PostStatusRemoved,
 		PostStatusCompleted}
 
-	if searchText != nil {
+	if filter.SearchText != nil {
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/models/post.go]: Good

File: application/models/post.go [hunk 3]
```diff
@@ -793,23 +801,27 @@ func (p *Posts) FindByUser(ctx context.Context, user User, destination, origin *
...
 		selectClause = selectClause + " AND (LOWER(title) LIKE ? or LOWER(description) LIKE ?)"
-		likeText := "%" + strings.ToLower(*searchText) + "%"
+		likeText := "%" + strings.ToLower(*filter.SearchText) + "%"
 		args = append(args, likeText, likeText)
 	}
+	if filter.PostID != nil {
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/models/post.go]: Good

File: application/models/post.go (hunk 1 above)
Comment (Schparky) [reply application/models/post.go]: The addition of the post ID was @fillup 's idea. I started by searching through all visible posts in a Go loop.

File: application/models/message_fixtures_test.go [hunk 4]
```diff
@@ -201,11 +183,29 @@ func CreateMessageFixtures_AfterCreate(ms *ModelSuite, t *testing.T) MessageFixt
 	posts[1].ProviderID = nulls.NewInt(users[0].ID)
 	ms.NoError(ms.DB.Save(&posts))
 
-	threads := []Thread{{PostID: posts[0].ID}, {PostID: posts[1].ID}}
+	threads := []Thread{
+		{
+			UUID:      domain.GetUUID(),
+			PostID:    posts[0].ID,
+			CreatedAt: time.Now().Add(-10 * time.Minute),
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/models/message_fixtures_test.go]: Good

File: application/models/message_fixtures_test.go [hunk 5]
```diff
@@ -51,46 +52,27 @@ func Fixtures_Message_GetSender(ms *ModelSuite, t *testing.T) MessageFixtures {
...
+//     Thread 1: User 0, User 2
+//     Thread 2: (no thread participants)
+// Post 2: Org A (no threads)
 func Fixtures_Message_Create(ms *ModelSuite, t *testing.T) MessageFixtures {
 	uf := createUserFixtures(ms.DB, 2)
-	org := uf.Organization
 	users := uf.Users
+	posts := createPostFixtures(ms.DB, 3, false)
 
-	location := Location{}
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/models/message_fixtures_test.go]: Yep, but some was moved to the AfterCreate test

File: application/models/message.go [hunk 6]
```diff
@@ -126,27 +144,44 @@ func (m *Message) GetThread() (*Thread, error) {
 	return &thread, nil
 }
 
-// Create a new message. Sends an `EventApiMessageCreated` event.
-func (m *Message) Create() error {
-	if err := create(m); err != nil {
-		return err
+// Create a new message if authorized.
+func (m *Message) Create(ctx context.Context, postUUID string, threadUUID *string, content string) error {
```
[the remark this answers is not in the document]
Comment (Schparky) [reply application/models/message.go]: I'm going to leave it -- I was think of some semblance of order of parameters.

**Pull request merged**